### PR TITLE
#3160 review/submission test data

### LINKED ISSUE #3160: [CAD/Project File Reviews] - Create Test Data for Submissions and Reviews
### Description

We want to have generic seed data that can be used for testing future endpoints/frontend features. This should consist of utils functions to create reviews and submissions, as well as a variety of different examples of submissions and reviews using these functions. Creating parts examples is in a different ticket, so make one sample part that can be used as the part field for all submission examples created.

### Acceptance Criteria

There should be a utils function to create reviews and submissions in the test-utils.ts file. There should then be examples made using those functions in a parts-review.ts file in the test-data folder. There should be a range of different examples including with and without optional fields, submissions without reviews, submissions with multiple reviews, and other applicable edge cases. Examples should have informative names.

### Proposed Solution

Create functions in test-utils.ts file that take in all fields for each model, and create prisma objects using that data. Then use these functions in parts-review.ts file in test-data folder to make a wide variety of examples. Look at other test data to see other examples. 

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by alexpark0**
## Changes

Created utils functions to make test data for part submissions and reviews, and used those functions to create seed data examples.

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #3160

### CODE REVIEW

File: src/backend/src/prisma/seed.ts [hunk 1]
```diff
@@ -2131,6 +2134,59 @@ const performSeed: () => Promise<void> = async () => {
     ner,
     false
   );
+
+  const partOnlyIdExample = await prisma.part.create({
+    data: {
+      partId: '001',
+      index: 0,
+      commonName: 'tire',
+      project: undefined,
```
Comment (chpy04) [inline src/backend/src/prisma/seed.ts]: Can you make this an actual project, preferably something already in the seed data

File: src/backend/src/prisma/seed.ts [hunk 2]
```diff
@@ -2131,6 +2134,59 @@ const performSeed: () => Promise<void> = async () => {
...
+      projectId: 'n/a',
+      userCreated: undefined,
```
Comment (chpy04) [inline src/backend/src/prisma/seed.ts]: Same thing here, this should be an actual user

File: src/backend/src/prisma/seed.ts [hunk 3]
```diff
@@ -2131,6 +2134,59 @@ const performSeed: () => Promise<void> = async () => {
...
     false
   );
+
+  const partOnlyIdExample = await prisma.part.create({
+    data: {
+      partId: '001',
+      index: 0,
+      commonName: 'tire',
+      project: undefined,
+      projectId: 'n/a',
```
Comment (chpy04) [inline src/backend/src/prisma/seed.ts]: This and userCreatedId below shouldn't need values if you do a connect for the actual relations

**Review by chpy04 — COMMENTED** [review]
Looks really good, just need a few minor changes

**Review by gcooper407 — APPROVED** [review]
Good changes, LGTM

File: src/backend/src/prisma/seed.ts [hunk 4]
```diff
@@ -2131,6 +2134,61 @@ const performSeed: () => Promise<void> = async () => {
...
+      project: {
+        connect: { projectId: project1Id }
+      },
+      userCreated: {
+        connect: { userId: batman.userId }
+      }
+    }
+  });
+
+  const emptyReviews: PartReview[] = [];
```
Comment (walker-sean) [inline src/backend/src/prisma/seed.ts]: No need to declare a const for this

File: src/backend/src/prisma/seed.ts [hunk 5]
```diff
@@ -2131,6 +2134,61 @@ const performSeed: () => Promise<void> = async () => {
...
+      fileIds: ['file1', 'file2'],
+      name: 'tire',
+      notes: 'black, round',
+      part: {
+        connect: { partId: partExample.partId }
+      },
+      userCreated: {
+        connect: { userId: batman.userId }
+      },
+      reviews: {
```
Comment (walker-sean) [inline src/backend/src/prisma/seed.ts]: You can just exclude this field if nothing is added

File: src/backend/src/prisma/seed.ts [hunk 6]
```diff
@@ -2131,6 +2134,61 @@ const performSeed: () => Promise<void> = async () => {
...
+      partReviewId: 'reviewId001',
+      fileIds: ['file1', 'file2'],
+      notes: 'this part submission sucks!!',
+      submission: {
+        connect: {
+          id: partSubmissionExample.id
+        }
+      },
+      popUps: {
+        connect: emptyPopups.map((popup) => ({ partReviewPopupId: popup.partReviewPopupId }))
```
Comment (walker-sean) [inline src/backend/src/prisma/seed.ts]: Same here

File: src/backend/tests/test-utils.ts [hunk 7]
```diff
@@ -671,3 +674,61 @@ export const createSlackMessageEvent = (
     ]
   };
 };
+
+export const createTestPartReview = async (
```
Comment (walker-sean) [inline src/backend/tests/test-utils.ts:735]: Why not use these?

**Review by walker-sean — CHANGES_REQUESTED** [review]
Can you add a few more objects?

File: src/backend/tests/test-utils.ts (hunk 7 above)
Comment (chpy04) [reply src/backend/tests/test-utils.ts:735]: Wanted to avoid using test file in seed data, this was the same issue that I made with Aryan

**Review by walker-sean — APPROVED** [review]
